test/handlers/UpgradeRegressionManager.handler.sol

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.30;
+
+// External Imports
+import {CommonBase} from "forge-std/Base.sol";
+import {StdCheats} from "forge-std/StdCheats.sol";
+import {StdUtils} from "forge-std/StdUtils.sol";
+
+// Internal Imports
+import {UpgradeRegressionManager} from "src/contracts/UpgradeRegressionManager.sol";
+import {FakeProtocolContract} from "test/fakes/FakeProtocolContract.sol";
+
+contract UpgradeRegressionManagerHandler is CommonBase, StdCheats, StdUtils {
+  /// @notice Enum to track the state of a rollback proposal
+  enum ProposalState {
+    NONE, // Not proposed
+    PROPOSED, // Proposed but not queued
+    QUEUED, // Queued but not executed
+    EXECUTED, // Successfully executed
+    CANCELLED // Cancelled
+
+  }
+
+  /// @notice Struct to track the state of a rollback proposal
+  struct RollbackProposal {
+    address[] targets;
+    uint256[] values;
+    bytes[] calldatas;
+    string description;
+    uint256 rollbackId;
+    ProposalState state;
+  }
+
+  UpgradeRegressionManager public urm;
+  FakeProtocolContract public target;
+
+  address public admin;
+  address public guardian;
+
+  // Mapping from rollbackId to proposal for O(1) lookup
+  mapping(uint256 => RollbackProposal) public proposals;
+
+  // Array to track all rollbackIds for iteration
+  uint256[] public rollbackIds;
+
+  string public constant DESCRIPTION = "test rollback";
+
+  constructor(UpgradeRegressionManager _urm, address _admin, address _guardian, FakeProtocolContract _target) {
+    // set initial state
+    urm = _urm;
+    admin = _admin;
+    guardian = _guardian;
+    target = _target;
+
+    // set up rollback
+    address[] memory _rollbackTargets = new address[](1);
+    uint256[] memory _rollbackValues = new uint256[](1);
+    bytes[] memory _rollbackCalldatas = new bytes[](1);
+
+    _rollbackTargets[0] = address(_target);
+    _rollbackValues[0] = 0;
+    _rollbackCalldatas[0] = abi.encodeWithSelector(FakeProtocolContract.setFee.selector, 10);
+
+    uint256 _rollbackId = urm.getRollbackId(_rollbackTargets, _rollbackValues, _rollbackCalldatas, DESCRIPTION);
+
+    vm.prank(admin);
+    try urm.propose(_rollbackTargets, _rollbackValues, _rollbackCalldatas, DESCRIPTION) {
+      proposals[_rollbackId] = RollbackProposal({
+        targets: _rollbackTargets,
+        values: _rollbackValues,
+        calldatas: _rollbackCalldatas,
+        description: DESCRIPTION,
+        rollbackId: _rollbackId,
+        state: ProposalState.PROPOSED
+      });
+      rollbackIds.push(_rollbackId);
+    } catch {}
+  }
+
```

Comment:
We didn't discuss this on the call, but we probably need a handler that simply warps the blockchain forward by some (bounded to something reasonable) random value.

Furthermore, it's probably worth considering the addition of a handler that calls `setRollbackQueueWindow` to some (bounded, reasonable) value at random as well.